api/src/main/java/com/datastrato/gravitino/rel/types/Types.java

```diff
@@ -1072,6 +1076,62 @@ public int hashCode() {
     }
   }
 
+  /** Represents a type that is defined in an external catalog. */
+  public static class ExternalType implements Type {
+    private final String catalogString;
+
+    /**
+     * Creates a new {@link ExternalType} with the given catalog string.
+     *
+     * @param catalogString The string representation of this type in the catalog.
+     * @return A new {@link ExternalType} with the given catalog string.
+     */
+    public static ExternalType of(String catalogString) {
```

Comment:
What is the meaning of `catalogString`, is this "catalog" the same meaning as "Catalog" object in Gravitino?

Reply:
For example, the data type `integer` in Gravitino corresponds to `int` MySQL, but `int4` in PostgreSQL.

Assuming we have not established a mapping for this type, we need to represent them using `External` type. When loading this type from the MySQL catalog, the `catalogString` is `integer`, while when loading it from PostgreSQL, its `catalogString` is `int4`.

Reply:
It is a bit weird to string `catalogString`, it is misleading and hard to relate to type system. Can you figure out a better name?

Reply:
Reference the naming of [Spark](https://github.com/apache/spark/blob/a29c9653f3d48d97875ae446d82896bdf0de61ca/sql/api/src/main/scala/org/apache/spark/sql/types/DataType.scala#L75). or named `sourceType`? `originalType`?